common/protob/messages-ripple.proto

```diff
@@ -29,23 +29,288 @@ message RippleAddress {
  * @next RippleSignedTx
  */
 message RippleSignTx {
-	repeated uint32 address_n = 1;				// BIP-32 path. For compatibility with other wallets, must be m/44'/144'/index'
-	optional uint64 fee = 2;					// fee (in drops) for the transaction
-	optional uint32 flags = 3;					// transaction flags
-	optional uint32 sequence = 4;				// transaction sequence number
-	optional uint32 last_ledger_sequence = 5;	// see https://developers.ripple.com/reliable-transaction-submission.html#lastledgersequence
-	optional RipplePayment payment = 6;			// Payment transaction type
+	// Common fields
+	repeated uint32 address_n = 1; // BIP-32 path. For compatibility with other wallets, must be m/44'/144'/index'
+	optional uint64 fee = 2; // fee (in drops) for the transaction
+	optional uint32 sequence = 3; // transaction sequence number
+	optional string account_txn_id = 4; // see https://xrpl.org/transaction-common-fields.html#accounttxnid
+	optional uint32 flags = 5; // transaction flags
+	optional uint32 last_ledger_sequence = 6; // see https://developers.ripple.com/reliable-transaction-submission.html#lastledgersequence
+	repeated RippleMemo memos = 7; // additional arbitrary information used to identify this transaction
+	repeated RippleSigner signers = 8; // array of objects that represent a multi-signature
+	optional uint32 source_tag = 9; // identify the reason for this payment, or a sender on whose behalf this transaction is made.
+
+
+
+	// Transaction fields
+	oneof transaction_data {
```

Comment:
Let's not use oneof. This is not supported.

Just add

```
optional Foo bar = ...
optional FooFoo barbar = ...
```

Reply:
I've changed all occurrences of oneof!